mars/services/task/tests/test_service.py

```diff
@@ -459,30 +457,93 @@ def f1(count: int):
 
         subtask_ids = set()
         for subtask in subtask_details.get('subtasks'):
-            assert ((subtask.get('status') == SubtaskStatus.pending.value) or (subtask.get('subtaskProgress') == -1 and subtask.get('status') == -1))
-            assert ((subtask.get('subtaskProgress') == 0) or (subtask.get('subtaskProgress') == -1 and subtask.get('status') == -1))
             assert subtask.get('subtaskId') not in subtask_ids
             subtask_ids.add(subtask.get('subtaskId'))
 
         for dependency in subtask_details.get('dependencies'):
             assert dependency.get('fromSubtaskId') in subtask_ids
             assert dependency.get('toSubtaskId') in subtask_ids
 
+@pytest.mark.asyncio
+async def test_get_subtask_detail(start_test_service):
+    sv_pool_address, task_api, storage_api = start_test_service
+    session_api = await SessionAPI.create(address=sv_pool_address)
+    ref = await session_api.create_remote_object(
+        task_api._session_id, 'progress_controller', _ProgressController)
+
+    def a():
+        return md.DataFrame([[1, 2], [3, 4]])
+
+    def b():
+        return md.DataFrame([[1, 2, 3, 4], [4, 3, 2, 1]])
+
+    def c(a, b):
+        return a.sum() * b.sum() / (a.product() * a.sum()) * b.product()
+
+    ra = mr.spawn(a)
+    rb = mr.spawn(b)
+    rc = mr.spawn(c, args=(ra, rb))
+
+    graph = TileableGraph([rc.data])
+    next(TileableGraphBuilder(graph).build())
+
+    task_id = await task_api.submit_tileable_graph(graph, fuse_enabled=True)
+
+    tileable_graph = await task_api.get_tileable_graph_as_json(task_id)
+    for tileable in tileable_graph.get('tileables'):
+        subtask_details = await task_api.get_tileable_subtask_detail(task_id, tileable.get('tileableId'))
+
+        subtask_ids = set()
+        for subtask_id, subtask_detail in subtask_details.items():
+            assert ((subtask_detail.get('status') >= 0 and subtask_detail.get('status') <= 4) or (subtask_detail.get('status') == -1 and subtask_detail.get('progress') == -1) or ((subtask_detail.get('status') == -2 and subtask_detail.get('progress') == -2)))
```

Comment:
This line is tooo loong. Try split this in multiple lines. It is also confusing using values of enumerations instead of values themselves, may use `SubtaskStatus.cancelled.value` instead.